other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java

```diff
@@ -1,76 +1,111 @@
+        List<TimeSeriesPoint> timeSeriesPointList = new ArrayList<>();
+        JSONArray seriesArray = requestJson.getJSONArray("series");
+        for (int i = 0; i < seriesArray.length(); i++) {
+            JSONObject seriesObject = seriesArray.getJSONObject(i);
+            TimeSeriesPoint timeSeriesPoint = new TimeSeriesPoint()
+                    .setTimestamp(OffsetDateTime.ofInstant(Instant.parse(seriesObject.getString("timestamp")), ZoneOffset.UTC))
+                    .setValue(seriesObject.getFloat("value"));
+            timeSeriesPointList.add(timeSeriesPoint);
         }
-        JSONTokener tokener = new JSONTokener(is);
-        String content = new JSONObject(tokener).toString();
-        CloseableHttpClient client = HttpClients.createDefault();
-        HttpPost request = new HttpPost(uriBase);
-
-        // Request headers.
-        request.setHeader("Content-Type", "application/json");
-        request.setHeader("Ocp-Apim-Subscription-Key", subscriptionKey);
-
-        try {
-            StringEntity params = new StringEntity(content);
-            request.setEntity(params);
-
-            CloseableHttpResponse response = client.execute(request);
-            try {
-                HttpEntity respEntity = response.getEntity();
-                if (respEntity != null) {
-                    System.out.println("----------");
-                    System.out.println(response.getStatusLine());
-                    System.out.println("Response content is :\n");
-                    System.out.println(EntityUtils.toString(respEntity, "utf-8"));
-                    System.out.println("----------");
+
+        DetectRequest request = new DetectRequest()
+                .setGranularity(TimeGranularity.fromString(requestJson.getString("granularity")))
+                .setSeries(timeSeriesPointList);
+
+        asyncClient.detectLastPointWithResponse(request).subscribe(response -> {
+            System.out.println("Request detail information...................................");
+            System.out.println("**************************************************************");
+            System.out.println(String.format("Request url is : %s", response.getRequest().getUrl().toString()));
+            response.getRequest().getHeaders().forEach(httpHeader -> {
+                System.out.println("**************************************************************");
+                System.out.println(String.format("Request httpHeader name is : %s", httpHeader.getName()));
+                System.out.println(String.format("Request httpHeader Value is : %s", httpHeader.getValue()));
+            });
+            response.getRequest().getBody().subscribe(byteBuffer -> {
+                System.out.println("**************************************************************");
+                Charset charset = Charset.forName("UTF-8");
```

Comment:
StandardCharsets.UTF_8 is more common.

Reply:
The process was deleted because the invoke method changed to [detectlastpoint].